int32 allied_type
---
float32 saturation
int32 error